### PR TITLE
use sourceEvent.currentTarget

### PR DESCRIPTION
**Pull request by Fil**
build https://files-dwh95j9sg.vercel.app/d3-selection.js

### CODE REVIEW

File: src/pointer.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import sourceEvent from "./sourceEvent.js";
 
-export default function(event, node = event.currentTarget) {
+export default function(event, node) {
   event = sourceEvent(event);
-  if (node) {
+  if (node || (node = event.currentTarget)) {
```
Comment (mbostock) [inline src/pointer.js:5]:
```suggestion
  if (node === undefined) node = event.currentTarget;
  if (node) {
```

File: src/pointers.js [hunk 2]
```diff
@@ -4,6 +4,7 @@ import sourceEvent from "./sourceEvent.js";
 export default function(events, node) {
   if (events.target) { // i.e., instanceof Event, not TouchList or iterable
     events = sourceEvent(events);
+    if (!node) node = events.currentTarget;
```
Comment (mbostock) [inline src/pointers.js:7]:
```suggestion
    if (node === undefined) node = events.currentTarget;
```

**Comment by Fil** [comment]
I've applied this to the “[Attitude Zooming](https://observablehq.com/d/7a912fba32fbe641)” notebook: all calls are now d3.pointers(event, that) on whatever the event is (in our case, a zoom event).

Comparison https://observablehq.com/compare/7a912fba32fbe641@507...7a912fba32fbe641@512:

Capture d’écran 2020-08-04 à 10 57 08

(Note that we have to be explicit about the target node (*that* in the code above). We can't rely on the event's currentTarget because the "zoom" event is a mousemove based on the view (Window), whereas the start event is a mousedown based on the real target (the canvas).)

**Comment by Fil** [comment]
The second notebook I had difficulties with was https://observablehq.com/d/d7b323beb3d4505a ; this PR solves it nicely too.

File: README.md [hunk 3]
```diff
@@ -680,14 +680,22 @@ If *parameters* is a function, it is evaluated for each selected element, in ord
...
 
 If the *target* is an HTML element, the event’s coordinates are translated relative to the top-left corner of the *target*’s [bounding client rectangle](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect). (As such, the coordinate system can only be translated relative to the client coordinates. See also [GeometryUtils](https://www.w3.org/TR/cssom-view-1/#the-geometryutils-interface).)
 
 Otherwise, [*event*.pageX, *event*.pageY] is returned.
 
+In the case of touch events, the coordinates of the first touch are returned. To handle multitouch interactions, see [pointers](#pointers) instead.
+
+<a name="pointers" href="#pointers">#</a> d3.<b>pointers</b>(<i>event</i>[, <i>target</i>]) · [Source](https://github.com/d3/d3-selection/blob/master/src/pointers.js)
+
+Returns a list [[*x0*, *y0*], [*x1*, *y1*]…] of coordinates of the specified *event*’s pointer locations relative to the specified *target*. For mouse, stylus or single touch events, [*x0*, *y0*] is equivalent to d3.pointer(event, target). In the case of multi-touch events, the returned array contains a pair of coordinates for each of the touches.
```
Comment (mbostock) [inline README.md:697]:
```suggestion
Returns an array [[*x0*, *y0*], [*x1*, *y1*]…] of coordinates of the specified *event*’s pointer locations relative to the specified *target*. For mouse, stylus or single touch events, [*x0*, *y0*] is equivalent to d3.pointer(event, target). In the case of multi-touch events, the returned array contains a pair of coordinates for each of the touches.

If *target* is not specified, it defaults to the source event’s currentTarget property, if available.
```

**Pull request merged**